crates/swap_provider/src/lib.rs

```diff
@@ -0,0 +1,13 @@
+use async_trait::async_trait;
+
+pub const SWAP_SLIPPAGE: f32 = 0.01;
+pub type SwapError = Box<dyn std::error::Error + Send + Sync>;
+
+#[async_trait]
+pub trait SwapProvider {
```

Comment:
add method to return supported chains to construct routing path easier?

Reply:
yes, let me add it